src/services/lease-service/routes/applicants.ts

```diff
@@ -447,13 +444,26 @@ export const routes = (router: KoaRouter) => {
           return
         }
 
-        const subjectHasHousingContractInSamePropertyAsListing =
-          await doesTenantHaveHousingContractInSamePropertyAsListing(
-            contact.data,
+        const validatableLease =
```

Comment:
Probably something I'm missing here but why is it enough to just check if `currentHousingContract` or `upcomingHousingContract` has values? Do we not need to check the contracts and the lease real estate code to determine if the housing contract and the lease matches?

Reply:
Do you mean `validatableLease`?

Before, we sent a parameter `{ currentHousingContract: Lease | undefined, upcomingHousingContract: Lease | undefined }` to `doesTenantHaveHousingContractInSamePropertyAsListing`. Then, inside that function, we returned "false" if neither of these existed.

The way I see it, the change here is to extract that "current OR upcoming OR undefined" from the function and put it here instead. Therefore, we use `validatableLease` to find the lease to validate. If there is none, it's treated the way it was before, like the tenant doesn't have either current or upcoming lease.

Reply:
Right! I got confused there but with this clarification I now understand